### DISCUSSION TITLE
Question about the abc parametrization across the two papers

### DISCUSSION THREAD

**Discussion by thomasdulacatos** [microsoft/mup#34] [General]
Hi. First, thanks for your work. 

Reading your two papers (https://arxiv.org/pdf/2011.14522.pdf and https://arxiv.org/pdf/2203.03466.pdf), I've seen some differences that I don't understand. On the first paper, you define the abc-Parametrization, and you write (page 2): 
![image](https://user-images.githubusercontent.com/100565522/211493557-84b3a7fa-da7f-4cb3-9d84-097fda014d13.png)
On page 9, you specified the Mup parametrization c is equal to 0, so the learning rate is not proportional to n.

But, on your second paper, page 5, when specifying the Mup parametrization, the learning rate are not the same for the different layers, and some are proportional to n: 
![image](https://user-images.githubusercontent.com/100565522/211494094-ca092a29-e04d-4a53-b3be-1204c47d3083.png)

I don't understand the reason of those differences ... Thanks for your time

**Comment by thegregyang** [comment]
Hi Thomas,

Thanks for your interest.

These two descriptions are equivalent. The [former paper (TP4)](https://arxiv.org/pdf/2011.14522.pdf) notes the following symmetry
<img width="919" alt="image" src="https://user-images.githubusercontent.com/53244851/211682690-d21934b3-c08f-495e-8072-d84e4e647cc9.png">
which remains true if you replace c with c_l (learning rate scaling for individual layer l) above. In other words, in the (a_l, b_l, c_l) for each layer, there is a 1-dimensional redundancy. Up to this redundancy, the two descriptions are the same.

The reason we differ in our presentation is that the former paper (TP4) is aimed at theoretists who tend to think about gradient descent with no "ad hoc" modification to the gradient (like per-layer learning rate), so we use a single learning rate for the entire network, at the expense of needing to scale multipliers (the `a_l`s) for different layers. However, the latter paper (TP5) is aimed at practitioners, who are very comfortable with per-layer learning rate, for example, through parameter groups in PyTorch optimizers. In contrast, they are less comfortable with changing model definition by inserting and changing multipliers. So we opt to absorb all multipliers into per-layer learning rates.

More details are given in the [latter paper (TP5)](https://arxiv.org/pdf/2203.03466.pdf):
<img width="1067" alt="image" src="https://user-images.githubusercontent.com/53244851/211683407-929e1160-544a-4fd4-83c6-83f60ac77c27.png">
<img width="1079" alt="image" src="https://user-images.githubusercontent.com/53244851/211683459-1f7bd102-7eb3-495d-bc20-2367dea0ace6.png">
